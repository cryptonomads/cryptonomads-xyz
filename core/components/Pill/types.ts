import { PillVariant } from './constants';

export interface PillProps {
  variant: PillVariant;
}
