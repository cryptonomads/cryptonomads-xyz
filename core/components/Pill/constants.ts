export enum PillVariant {
  INFO = 'info',
  SUCCESS = 'success',
  WARNING = 'warning',
  DANGER = 'danger',
}
