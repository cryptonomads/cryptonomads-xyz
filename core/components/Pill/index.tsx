export { Pill as default } from './Pill';
export { PillVariant } from './constants';
