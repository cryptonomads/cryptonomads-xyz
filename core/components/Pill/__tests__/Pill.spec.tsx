import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Pill from '../';
import { PillVariant } from '../constants';

afterEach(() => {
  cleanup();
});

describe('Pill', () => {
  it('Renders the Pill component with the info variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant={PillVariant.INFO}>test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the success variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant={PillVariant.SUCCESS}>test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the danger variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant={PillVariant.DANGER}>test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders the Pill component with the warning variant', () => {
    const { getByText, asFragment } = render(
      <Pill variant={PillVariant.WARNING}>test</Pill>
    );
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
