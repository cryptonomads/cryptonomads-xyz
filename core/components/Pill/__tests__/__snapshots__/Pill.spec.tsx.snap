// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Renders the Pill component with the danger variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  border-radius: var(--border-radius-1);
  height: 28px;
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 5px 8px 5px 8px!important;
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: var(--maximeheckel-colors-danger-emphasis);
  color: var(--maximeheckel-colors-danger);
}

<span
    class="emotion-0 emotion-1"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the info variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  border-radius: var(--border-radius-1);
  height: 28px;
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 5px 8px 5px 8px!important;
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: var(--maximeheckel-colors-emphasis);
  color: var(--maximeheckel-colors-brand);
}

<span
    class="emotion-0 emotion-1"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the success variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  border-radius: var(--border-radius-1);
  height: 28px;
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 5px 8px 5px 8px!important;
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: var(--maximeheckel-colors-success-emphasis);
  color: hsl(var(--palette-green-80));
}

<span
    class="emotion-0 emotion-1"
  >
    test
  </span>
</DocumentFragment>
`;

exports[`Pill Renders the Pill component with the warning variant 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  border-radius: var(--border-radius-1);
  height: 28px;
  display: -webkit-inline-box!important;
  display: -webkit-inline-flex!important;
  display: -ms-inline-flexbox!important;
  display: inline-flex!important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 5px 8px 5px 8px!important;
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background: var(--maximeheckel-colors-warning-emphasis);
  color: hsl(var(--palette-orange-80));
}

<span
    class="emotion-0 emotion-1"
  >
    test
  </span>
</DocumentFragment>
`;
